tests/unit/constraints/test_tabular.py

```diff
@@ -5188,6 +5188,128 @@ def test_is_valid(self):
         expected = pd.Series([True, False, True, False, True, True])
         pd.testing.assert_series_equal(valid, expected)
 
+    def test_is_valid_custom_index_same_values(self):
+        """Test the ``Unique.is_valid`` method.
+
+        This method should return a pd.Series where the index
+        of the first occurence of a unique combination of ``instance.columns``
+        is set to ``True``, and every other occurence is set to ``False``.
+
+        Input:
+        - DataFrame with multiple of the same combinations of columns.
+        - DataFrame has a custom index column which is set to 0 for rows.
+        Output:
+        - Series with the index of the first occurences set to ``True``.
+        Github Issue:
+        - Problem is described in: https://github.com/sdv-dev/SDV/issues/616
+        """
+        # Setup
+        instance = Unique(columns=['a', 'b', 'c'])
+
+        # Run
+        data = pd.DataFrame({
+            'a': [1, 1, 2, 2, 3],
+            'b': [5, 5, 6, 6, 7],
+            'c': [8, 8, 9, 9, 10]
+        }, index=[0, 0, 0, 0, 0])
+        valid = instance.is_valid(data)
+
+        # Assert
+        expected = pd.Series([True, False, True, False, True], index=[0, 0, 0, 0, 0])
+        pd.testing.assert_series_equal(valid, expected)
+
+    def test_is_valid_custom_index_not_sorted(self):
+        """Test the ``Unique.is_valid`` method.
+
+        This method should return a pd.Series where the index
+        of the first occurence of a unique combination of ``instance.columns``
+        is set to ``True``, and every other occurence is set to ``False``.
+
+        Input:
+        - DataFrame with multiple of the same combinations of columns.
+        - DataFrame has a custom index column which is set in an unsorted way.
+        Output:
+        - Series with the index of the first occurences set to ``True``.
+        Github Issue:
+        - Problem is described in: https://github.com/sdv-dev/SDV/issues/617
+        """
+        # Setup
+        instance = Unique(columns=['a', 'b', 'c'])
+
+        # Run
+        data = pd.DataFrame({
+            'a': [1, 1, 2, 2, 3],
+            'b': [5, 5, 6, 6, 7],
+            'c': [8, 8, 9, 9, 10]
+        }, index=[2, 1, 3, 5, 4])
```

Comment:
It seems like this test and the one above both test the same combination of columns, and also that the original index is preserved. Is it enough just to have one of these tests? Is having the index set to 0 a special case?

Reply:
I think it is a very similar test, since the indices of `pandas` do not have to be unique.
I created the test from the Issues to confirm that this problem is reproducible.

I would probably leave it as is because I like to see problems specified in Issues created as tests.